test/modules/packages/linearalgebra/correctness/test_identities1.chpl

```diff
@@ -0,0 +1,168 @@
+    rng.fillRandom(A);
+    rng.fillRandom(B);
+    rng.fillRandom(C);
+    var r1 = dot(A,dot(B,C));
+    var r2 = dot(dot(A,B),C);
+    var t1 = max reduce abs(r1/r2); 
+    if verbose && (ii==0) {
+      writef("%s : %n, %n\n",testName, t1, t1-1.0);
+    }
+    testRelCompare(t1,1.0,eps,testName);
+  }
+  if verbose {
+    writef("Test %s passed!\n", testName);
+  }
+}
+
+/* (AB)^t = B^t A^t */
+// Do an elementwise comparison
+proc test_transpose_product(type t) {
+  const testName="transpose_product [%s]".format(t:string);
+  const eps = threshold(t);
+  var A = Matrix(L, M, t);
+  var B = Matrix(M, N, t);
+
+  var rng=new RandomStream(t, seed=seed);
+  for ii in 0.. #nIters {
+    rng.fillRandom(A);
+    rng.fillRandom(B);
+    var r1 = transpose(dot(A,B));
+    var r2 = dot(transpose(B),transpose(A));
+    var t1 = max reduce abs(r1/r2); 
+    if verbose && (ii==0) {
+      writef("%s : %n, %n\n",testName, t1, t1-1.0);
+    }
+    testRelCompare(t1,1.0,eps,testName);
+  }
+  if verbose {
+    writef("Test %s passed!\n", testName);
+  }
+}
+
+/* tr(AB) = tr(BA) */
+proc test_trace_rotate(type t) {
+  const testName="trace_rotate [%s]".format(t:string);
+  const eps = threshold(t);
+  var A = Matrix(M, N, t);
+  var B = Matrix(N, M, t);
+
+  var rng=new RandomStream(t, seed=seed);
+  for ii in 0.. #nIters {
+    rng.fillRandom(A);
+    rng.fillRandom(B);
+    var t1 = trace(dot(A,B));
+    var t2 = trace(dot(B,A));
+    if verbose && (ii==0) {
+      writef("%s : %n, %n\n",testName, t1,t2);
+    }
+    testRelCompare(t1,t2,eps,testName);
+  }
+  if verbose {
+    writef("Test %s passed!\n", testName);
+  }
+}
+
+/* vAw = tr(Awv) */
+proc test_trace_vector_rotate(type t) {
+  const testName="trace_vector_rotate [%s]".format(t:string);
+  const eps = threshold(t);
+  var A = Matrix(M, M, t);
+  var v = Vector(M, t);
+  var w = Vector(M, t);
+
+  var rng=new RandomStream(t, seed=seed);
+  for ii in 0.. #nIters {
+    rng.fillRandom(A);
+    rng.fillRandom(v);
+    rng.fillRandom(w);
+    var t1 = dot(dot(A,w),v);
+    var t2 = trace(dot(A, outer(w,v)));
+    if verbose && (ii==0) {
+      writef("%s : %n, %n\n", testName, t1,t2);
+    }
+    testRelCompare(t1,t2,eps,testName);
+  }
+  if verbose {
+    writef("Test %s passed!\n", testName);
+  }
+}
+
+//// Helpers
+proc threshold(type t) {
+  select t {
+    when real(32) do return singleThresh;
+    when real(64) do return doubleThresh;
+    when complex(64) do return singleThresh;
+    when complex(128) do return doubleThresh;
+    otherwise halt("Unknown type!");
+    }
+}
+
+// Relative comparisons -- good when expected != 0
+proc testRelCompare(expected, actual, thresh : real,testname:string) {
+  var diff = abs(actual/expected - 1.0);
+  if diff > thresh {
+    writef("Failure in %s : %er \n",testname, diff);
+  }
+}
+
+// Absolute comparisons
+proc testAbsCompare(expected, actual, thresh : real,testname:string) {
+  var diff = abs(actual - expected);
+  if diff > thresh {
```

Comment:
@monsij --- Sorry to be slow here, but I'm not sure what you are referring to. The function above is simply a helper function to compare a result to its expected value and print out an appropriate message if the comparison fails (given some tolerance). Mostly to avoid copying the same logic everywhere.....

Maybe these should be renamed to ``absCompare`` and ``relCompare`` to be clearer.... I'm no longer sure why I chose those names...